### PR TITLE
breadcrumb works without parent or grandparent

### PR DESCRIPTION
**Pull request by yuna-iwata**
- if parent/grandparent is null, set parent and grandparent to a state where the name is "" and slug is the current page

### CODE REVIEW

File: web/src/Pages/TopicPage/BreadCrumbs/Breadcrumbs.jsx [hunk 1]
```diff
@@ -1,73 +1,98 @@
...
 import { Link } from "react-router-dom";
-import { React } from "react";
+import { React, useEffect, useState } from "react";
 import PropType from "prop-types";
 import "../../../Styling/TopicBreadcrumbs/TopicBreadcrumbs.css";
 import { sendBreadCrumbEvent } from "../../../utils/gaEvents";
 
 function Breadcrumbs({ grandParent, parent, current }) {
+  const [dad, setDad] = useState(parent);
+  const [grandad, setGrandad] = useState(grandParent);
```
Comment (DomVinyard) [inline web/src/Pages/TopicPage/BreadCrumbs/Breadcrumbs.jsx:10]: 😆 nice

Comment (s-dilip) [inline web/src/Pages/TopicPage/BreadCrumbs/Breadcrumbs.jsx:10]: Could perhaps use better names.
```suggestion
  const [daddy, setDaddy] = useState(parent);
  const [grandaddy, setGrandaddy] = useState(grandParent);
```

Comment (yuna-iwata) [reply web/src/Pages/TopicPage/BreadCrumbs/Breadcrumbs.jsx:10]: ah, that is a good idea but i raise you:
```suggestion
  const [mommy, setMommy] = useState(parent);
  const [grandmommy, setGrandMommy] = useState(grandParent);
```

**Review by mdapim — APPROVED** [review]
Works well on deploy, looks good

File: web/src/Pages/TopicPage/BreadCrumbs/Breadcrumbs.jsx (hunk 1 above)
Comment (DomVinyard) [reply web/src/Pages/TopicPage/BreadCrumbs/Breadcrumbs.jsx:10]: love it

**Review by DomVinyard — APPROVED** [review]

**Pull request merged**